src/anyp/Uri.cc

```diff
@@ -167,6 +167,20 @@ urlParseProtocol(const char *b)
     return AnyP::PROTO_NONE;
 }
 
+bool
+urlAppendDomain(char *host)
+{
+    if (Config.appendDomain && !strchr(host, '.') && !strchr(host, ':')) {
+        const auto dlen = strlen(host);
+        if (dlen > (SQUIDHOSTNAMELEN - Config.appendDomainLen - 1)) {
```

Comment:
This expression is both 
* more difficult to read/interpret than necessary and
* potentially dangerous with regard to unsigned values wrapping when (unvalidated) `Config.appendDomainLen` exceeds `SQUIDHOSTNAMELEN`

I hope this version addresses both problems:
```suggestion
        if (dlen + Config.appendDomainLen > SQUIDHOSTNAMELEN - 1) {
```

The `-1` crap is unfortunate, but I am guessing we have to live with it because some code assumes that `SQUIDHOSTNAMELEN` includes a c-string terminator. You can move that `1` to the left side if you prefer. IMHO, it is more logical on the right side because it is derived from SQUIDHOSTNAMELEN _meaning_ (is not dependent on input) and affects the _limit_, which is what the right side should be about.

N.B. We could also protect from `dlen + Config.appendDomainLen` overflows but, unlike underflows, those are just too unlikely to happen in real deployments. Your call though.

Reply:
The (SQUIDHOSTNAMELEN - Config.appendDomainLen) underflow is the less likely to be problematic. That is only run-time variable with the append_domain length. Which is highly unlikely any admin will exceed (and we can validate on configure if you want to be completely certain). Whereas the suggested (dlen+ N) form is dependent on arbitrary remote input. Using an explicit int64 though should be sufficient until this function is dropped entirely by the SBuf refactoring.

Reply:
I will not waste time arguing which of the two unlikely events is more likely. Fixing the expression while preventing sum overflows using 64bit integers is, naturally, fine with me. If possible, please avoid unsigned-to-signed conversion though: Use `uint64_t` instead of `int64_t`.